diff --git a/build/cjs/components/Inputs/SlateEditor/SlateEditor.interface.d.ts b/build/cjs/components/Inputs/SlateEditor/SlateEditor.interface.d.ts
index e72a23657998f5c45b1fee57ddebcca662129a3b..d01daadffe749b6b130a7d0e5040f1634e4490bd 100644
--- a/build/cjs/components/Inputs/SlateEditor/SlateEditor.interface.d.ts
+++ b/build/cjs/components/Inputs/SlateEditor/SlateEditor.interface.d.ts
@@ -50,12 +50,4 @@ export interface ISlateEditor extends FormControlProps<Descendant[]>, Omit<Edita
         editable: React.ReactNode;
     }) => React.ReactNode;
 }
-declare module 'slate' {
-    interface CustomTypes {
-        Text: Text;
-        Range: Range;
-        Editor: Editor;
-        Element: AnyElement;
-    }
-}
 //# sourceMappingURL=SlateEditor.interface.d.ts.map
\ No newline at end of file
diff --git a/build/esm/components/Inputs/SlateEditor/SlateEditor.interface.d.ts b/build/esm/components/Inputs/SlateEditor/SlateEditor.interface.d.ts
index e72a23657998f5c45b1fee57ddebcca662129a3b..d01daadffe749b6b130a7d0e5040f1634e4490bd 100644
--- a/build/esm/components/Inputs/SlateEditor/SlateEditor.interface.d.ts
+++ b/build/esm/components/Inputs/SlateEditor/SlateEditor.interface.d.ts
@@ -50,12 +50,4 @@ export interface ISlateEditor extends FormControlProps<Descendant[]>, Omit<Edita
         editable: React.ReactNode;
     }) => React.ReactNode;
 }
-declare module 'slate' {
-    interface CustomTypes {
-        Text: Text;
-        Range: Range;
-        Editor: Editor;
-        Element: AnyElement;
-    }
-}
 //# sourceMappingURL=SlateEditor.interface.d.ts.map
\ No newline at end of file
diff --git a/build/next/components/Inputs/SlateEditor/SlateEditor.interface.d.ts b/build/next/components/Inputs/SlateEditor/SlateEditor.interface.d.ts
index e72a23657998f5c45b1fee57ddebcca662129a3b..d01daadffe749b6b130a7d0e5040f1634e4490bd 100644
--- a/build/next/components/Inputs/SlateEditor/SlateEditor.interface.d.ts
+++ b/build/next/components/Inputs/SlateEditor/SlateEditor.interface.d.ts
@@ -50,12 +50,4 @@ export interface ISlateEditor extends FormControlProps<Descendant[]>, Omit<Edita
         editable: React.ReactNode;
     }) => React.ReactNode;
 }
-declare module 'slate' {
-    interface CustomTypes {
-        Text: Text;
-        Range: Range;
-        Editor: Editor;
-        Element: AnyElement;
-    }
-}
 //# sourceMappingURL=SlateEditor.interface.d.ts.map
\ No newline at end of file
diff --git a/src/components/Inputs/SlateEditor/SlateEditor.interface.ts b/src/components/Inputs/SlateEditor/SlateEditor.interface.ts
index bc9a905b1e8f1e051ae3e5f75cf07957eebd6943..f309bfd300916fa3636e77522a0c1ad61b925b33 100644
--- a/src/components/Inputs/SlateEditor/SlateEditor.interface.ts
+++ b/src/components/Inputs/SlateEditor/SlateEditor.interface.ts
@@ -65,12 +65,3 @@ export interface ISlateEditor
   pluginsOptions?: PluginsOptions;
   editableContainer?: (props: { editable: React.ReactNode }) => React.ReactNode;
 }
-
-declare module 'slate' {
-  interface CustomTypes {
-    Text: Text;
-    Range: Range;
-    Editor: Editor;
-    Element: AnyElement;
-  }
-}
